odoo/service/server.py

```diff
@@ -332,6 +333,18 @@ def close_socket(self, sock):
                 raise
         sock.close()
 
+    def atexit(self, func):
+        """ Register a cleanup function to be executed when the server stops """
+        self._atexit_funcs.append(func)
+
+    def _run_atexit_funcs(self):
+        for func in self._atexit_funcs:
+            try:
+                func()
+            except Exception:
+                _logger.warning("Could not execute atexit function", ext_info=True)
```

Comment:
Can we instanciate a server, `start()` it, then `stop()` it, then `start()` it again?
If so, the cleanup should empty the list `self._atexit_funcs`.

Reply:
No, the cycle is start -> stop (-> optional reexec via `os.execve` which spawn a new process)